sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -78,7 +80,41 @@ private string GetSessionFilePath()
             string className = testAdapter.ClassName.Substring(testAdapter.ClassName.LastIndexOf('.') + 1);
 
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
-            return Path.Combine(TestContext.CurrentContext.TestDirectory,
+
+            var path = TestContext.CurrentContext.TestDirectory;
+            var newpath = path.Substring(0, path.IndexOf("artifacts")) + "sdk";
+
+            // check if Track 1 or Track 2
+            int firstIndex = path.IndexOf("bin\\") + "bin\\".Length;
+            int lastIndex = path.LastIndexOf(".Tests");
+            var testname = path.Substring(firstIndex, lastIndex - firstIndex);
```

Comment:
Handle exceptions here if it substrings are not in the path. Just put the files in the old location if the path is not structured as expected, since the dll would be running in an unexpected location.